tests/integration/deliver_grant_funding/test_routes.py

```diff
@@ -965,71 +965,107 @@ def test_grant_check_your_answers_post_creates_grant(authenticated_platform_admi
+
+
+def test_list_users_for_grant_with_member_no_add_member_button(
+    authenticated_member_client, templates_rendered, factories
+):
     grant = factories.grant.create()
     user = get_current_user()
     factories.user_role.create(user=user, role=RoleEnum.MEMBER, grant=grant)
-    result = authenticated_member_client.get(
-        url_for(
-            "deliver_grant_funding.list_users_for_grant",
-            grant_id=grant.id,
-        ),
-    )
-    soup = BeautifulSoup(result.data, "html.parser")
-    button = soup.find("a", string=lambda text: text and "Add grant team member" in text)
-    assert button is None, "'Add grant team member' button is available"
-    headers = soup.find_all("th")
-    header_texts = [th.get_text(strip=True) for th in headers]
-    expected_headers = ["Email", "Role"]
-    for expected in expected_headers:
-        assert expected in header_texts, f"Header '{expected}' not found in table"
-    assert "Grant team" in soup.h1.text
+    response = authenticated_member_client.get(url_for("deliver_grant_funding.list_users_for_grant", grant_id=grant.id))
+    soup = BeautifulSoup(response.data, "html.parser")
+    assert soup.find("a", string=lambda text: text and "Add grant team member" in text) is None
 
 
-def test_add_user_to_grant_with_platform_admin(authenticated_platform_admin_client, factories):
-    # TODO this PR only consists the UI/UX changes & separate PR FSPT-528 will do the backend work
+def test_list_users_for_grant_with_member(authenticated_member_client, templates_rendered, factories):
     grant = factories.grant.create()
-    result = authenticated_platform_admin_client.get(
-        url_for(
-            "deliver_grant_funding.add_user_to_grant",
-            grant_id=grant.id,
-        ),
-    )
-    soup = BeautifulSoup(result.data, "html.parser")
-    assert "Add grant team member" in soup.h1.text
-    assert "What’s their email address?" in soup.h1.text
-    button = soup.find("button", string=lambda text: text and "Continue" in text)
-    assert button is not None, "'Continue' button not found"
-    button = soup.find("a", string=lambda text: text and "Back" in text)
-    assert button is not None, "'Back' button not found"
+    user = get_current_user()
+    factories.user_role.create(user=user, role=RoleEnum.MEMBER, grant=grant)
+    response = authenticated_member_client.get(url_for("deliver_grant_funding.list_users_for_grant", grant_id=grant.id))
+    soup = BeautifulSoup(response.data, "html.parser")
+    # UI assertions
+    assert soup.find("a", string=lambda text: text and "Add grant team member" in text) is None
+    assert all(header in [th.get_text(strip=True) for th in soup.find_all("th")] for header in ["Email", "Role"])
+    assert "Grant team" in soup.h1.text
+    # BE assertions
+    users = templates_rendered[0][1].get("grant").users
+    assert users and len(users) == 1
```

Comment:
You're a bit inconsistent about your assertions - sometimes you store the thing you're asserting against in a nice variable (eg `users`), and sometimes you don't.

I think we should prioritise readability in tests, so if we can get anything long and awkward not inlined, that'd be great. Example:

```suggestion
    # UI assertions
    add_grant_team_member_button = soup.find("a", string=lambda text: text and "Add grant team member" in text)
    headers = {th.get_text(strip=True) for th in soup.find_all("th")}
    assert add_grant_team_member_button is None
    assert headers == {"Email", "Name", "Role"}
    assert "Grant team" in soup.h1.text

    # BE assertions
    users = templates_rendered[0][1].get("grant").users
    assert users and len(users) == 1
```